fix(@desktop/general): clicking push notification does expand the app but does not open correct channel/chat

saledjenic opened the pull request:
Adding notifications for MacOS and Windows in an OS native way applied. For Windows we're referring to an already added HWND which is created adding the app to a system tray.

saledjenic committed:
chore(@desktop/general): CMake include files optimization

pinebit commented on lib/src/StatusNotification/StatusOSNotification.cpp:
```diff
@@ -0,0 +1,123 @@
+    tnd.cbSize = sizeof(NOTIFYICONDATA);
+    tnd.uVersion = NOTIFYICON_VERSION_4;
+
+    const char *cTitle = title.toLocal8Bit().data();
```
`toLocal8Bit()`:
`The returned byte array is undefined if the string contains characters not supported by the local 8-bit encoding.`
would this work for translated strings? maybe it worth considering unicode version if it exists?

pinebit commented on lib/src/StatusNotification/StatusOSNotification.cpp:
```diff
@@ -0,0 +1,123 @@
+    m_hwnd = nullptr;
+    initNotificationWin();
+#elif defined Q_OS_MACOS
+    m_notificationHelper = nullptr;
```
do we need to delete this object in destructor?

saledjenic committed:
fix(@desktop/general): (macos) clicking push notification does expand the app but does not open correct channel/chat

MacOS notifications handled in a native way.
This change is required as part of the fix for ticket 2996.

Fixes: #2996

saledjenic committed:
fix(@desktop/general): (windows) clicking push notification does expand the app but does not open correct channel/chat

Windows notifications added in a native way.
This change is required as part of the fix for ticket 2996.

Fixes: #2996

saledjenic commented:
@pinebit thanks for the comments, I arranged all them, now we are using [toWCharArray](https://doc.qt.io/qt-5.14/qstring.html#toWCharArray) method which deals with unicode well.
